src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,41 @@
 'use strict';
 
+const healthByDefault = 100;
+const decreaseOfHealth = 50;
+const deadBeast = 0;
+
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(name, health = healthByDefault) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name, health);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    const isCarnivore = animal instanceof Carnivore;
+
+    if (!isCarnivore && !animal.hidden) {
+      animal.health -= decreaseOfHealth;
+    }
+
+    if (animal.health <= deadBeast) {
+      Animal.alive = Animal.alive.filter((beast) => beast.health > deadBeast);
```

Comment:
The logic for removing dead animals from the `Animal.alive` array is correct, but it might be more efficient to directly remove the specific `animal` instance if its health is zero or below, rather than filtering the entire array. Consider using `Animal.alive = Animal.alive.filter(beast => beast !== animal)`.